agent/agent.go

```diff
@@ -305,67 +305,23 @@ func (a *Agent) loop(ctx context.Context) error {
 				}
```

Comment:
I would absorb `GetNextTask` into `processNextTask` as well, since the timer resetting logic is the same as the `ShouldResetTimer`/`ShouldContinue` case. It also seems like `processNextTask` is the only thing that uses `GetNextTask`'s output.

Reply:
If you don't have strong feelings about it, I would prefer to keep them separate, I think it helps with readability.